Dockerfile

```diff
@@ -1,14 +1,10 @@
 FROM python:3
 
-ENV UV_SYSTEM_PYTHON=true
-
 RUN apt-get update && apt-get install -y make
 
 WORKDIR /sdk
 
-COPY Makefile pyproject.toml requirements.txt requirements-dev.txt vars.mk ./
-
-RUN --mount=type=cache,mode=0755,target=/root/.cache/pip make deps
```

Comment:
Since the `make dep` step has been combined with `make dev`, the iterative development process of integrations is quite a bit slower. On main, it took 7.3s to run `make -C integration build`, while on this branch, it took 13.1s. This isn't a huge difference, but it is noticeable since this is part of my development workflow. The reasoning may be a bit nuanced...

Line 11 creates a cached layer with the dependencies. When the source code is changed, it invalidates line 14. Subsequently, only line 16, which is the editable install, is recomputed. So, when doing development work on the source code, I can make changes without forcing Docker to reinstall all dependencies.


How difficult would it be to separate back out the editable install (`pip install -e .`) from the general dependency installation?

This isn't a blocker, but I would prefer the faster iterative build times.

Reply:
🥳 Solved with `$(UV) sync --frozen --no-install-project`

Installs given the lock file (that is copied over). Only install the project dependencies, but not the `posit` package (which needs setuptools-scm and `./src` to be compiled).